p2p/pex/pex_reactor.go

```diff
@@ -752,10 +818,33 @@ func (r *Reactor) attemptDisconnects() {
 
 func markAddrInBookBasedOnErr(addr *p2p.NetAddress, book AddrBook, err error) {
 	// TODO: detect more "bad peer" scenarios
+	// TODO, blacklist peer if fmt.Errorf("peer is on a different network. Got %v, expected %v", other.Network, info.Network)
```

Comment:
This is an `ErrSwitchAuthenticationFailure` as below, isn't it?

Reply:
I don't think so right? The error I am referring to is defined as ErrDifferentNetwork, I don't see where it would get wrapped as ErrSwitchAuthenticationFailure. That error appears as "failed to authenticate peer. Dialed %s"